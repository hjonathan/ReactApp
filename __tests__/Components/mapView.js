import 'react-native';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import MapViewer from '../../App/Components/MapView';

describe('Component >>> mapView', () => {
    const defaultRegion = {
            latitude: null,
            longitude: null,
            latitudeDelta: 0.03901,
            longitudeDelta: 0.03003
        },
        event = {
            nativeEvent: {
                coordinate: null
            }
        },
        marker = {
            _component: {
                animateMarkerToCoordinate: (coord, time) => {}
            }
        },
        onError = () => {};

    let mapView,
        mapViewDefault;

    beforeEach(() => {
        mapViewDefault = shallow(<MapViewer />);
        mapView = shallow(<MapViewer onError={onError} />);
    });

    test('Render mapView', () => {
        expect(toJson(mapView)).toMatchSnapshot();
        expect(toJson(mapViewDefault)).toMatchSnapshot();
    });

    test('onPress', () => {
        mapView.instance().marker = marker;
        mapView.instance().onPress(event);
        expect(mapView.instance().state.coordinate).toEqual(null);
    });

    test('onChangeRegion', () => {
        mapView.instance().onChangeRegion(defaultRegion);
        expect(mapView.instance().state.region).toEqual(defaultRegion);
    });

    test('getLastPosition', () => {
        expect(mapView.instance().getLastPosition()).toEqual({latitude: null, longitude: null});
    });

    test('errorEvent', () => {
        expect(mapViewDefault.instance().errorEvent({})).toMatchSnapshot();
        expect(mapView.instance().errorEvent({})).toMatchSnapshot();
    });

    test('renderLocationButton', () => {
        let locationButton = mapView.instance().renderLocationButton();
        expect(locationButton.props.children.props.onPress()).toMatchSnapshot();
    });

    test('renderMap', () => {
        mapView.instance().setState({
            region: {
                latitude: -16.003,
                longitude: 68.003,
                latitudeDelta: 0.03901,
                longitudeDelta: 0.03003
            }
        });
        let map = mapView.instance().renderMap(),
            marker = map.props.children;
        expect(marker.ref()).toMatchSnapshot();
    });
});
