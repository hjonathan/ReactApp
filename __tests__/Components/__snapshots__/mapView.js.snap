// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> mapView Render mapView 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FCFCFC",
        "borderRadius": 2,
        "height": 38,
        "justifyContent": "center",
        "margin": 16,
        "position": "absolute",
        "right": 0,
        "width": 38,
        "zIndex": 2,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="gps-fixed"
        size={12}
        style={
          Object {
            "fontSize": 22,
          }
        }
      />
    </TouchableOpacity>
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#FFFDF4",
        "flex": 1,
      }
    }
  />
</Component>
`;

exports[`Component >>> mapView Render mapView 2`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FCFCFC",
        "borderRadius": 2,
        "height": 38,
        "justifyContent": "center",
        "margin": 16,
        "position": "absolute",
        "right": 0,
        "width": 38,
        "zIndex": 2,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="gps-fixed"
        size={12}
        style={
          Object {
            "fontSize": 22,
          }
        }
      />
    </TouchableOpacity>
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#FFFDF4",
        "flex": 1,
      }
    }
  />
</Component>
`;

exports[`Component >>> mapView errorEvent 1`] = `undefined`;

exports[`Component >>> mapView errorEvent 2`] = `undefined`;

exports[`Component >>> mapView renderLocationButton 1`] = `undefined`;

exports[`Component >>> mapView renderMap 1`] = `undefined`;
